Mathlib/SetTheory/Cardinal/Regular.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2018 Nir Paz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nir Paz
+-/
+import Mathlib.SetTheory.Cardinal.Cofinality
+
+/-!
+# Regular cardinals
+
+We define singular cardinals and prove theorems about singular and regular cardinals.
+
+## Main definitions and results
+
+* `Cardinal.IsSingular c` means that `c` is an infinite cardinal which is not regular. That is,
+  its cofinality is smaller than itself.
+-/
+
+
+open Ordinal Cardinal
+
+namespace Cardinal
+
+section Singular
+
+variable {c : Cardinal}
+
+/-- A cardinal is singular if it is infinite and not regular. -/
+def IsSingular (c : Cardinal) : Prop :=
+  ℵ₀ ≤ c ∧ ¬c.IsRegular
```

Comment:
This feels like it belongs right next to `IsRegular`. Would you mind either moving `IsSingular` to `SetTheory.Cardinal.Cofinality` or moving `IsRegular` to `SetTheory.Cardinal.Regular`?

Reply:
I mentioned a bunch of comments ago that I'd like to move all the stuff about `IsRegular`, `IsStrongLimit`, and `IsInaccesible` to its own file, but that I thought it'd be fine to merge this PR first and then move those definitions to this file afterwards.

Reply:
I tend to have the opposite opinion: split them first, then do this PR. This way, you don't have weird copyrights in the future